docs/plugin/lifecycle.md

```diff
@@ -239,6 +239,64 @@ void TestPlugin::Unregister(IStateControl::INotification* notification)
 }
 ```
 
+## Hibernate
+!!! note
+    `Hibernated` state is not avaible by default. To enable it proper option must be switch on while building Thunder.
+
+Hibernate is a state where plugin is not running and will not respond to requests. Memory of the plugin is flushed to disk and released. When a plugin goes into `Suspended` state it slows down the CPU usage but it still occupies memory. Hibernate allows memory recovery of plugins that are not currently in use. This can come in handy when there is very little available and we need to be careful not to use it all. As the memory must first be written to and then read from disk, this increases the recovery time of the plugin.
+
+ To put plugin in `Hibernated` state two conditions must be met:
+
+* Plugin must be `Activated`
+* Plugin must be `Suspended`
```

Comment:
There is a 3rd condition - the plugin must be running out of process (hopefully this is obvious but worth stating)